src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -552,6 +551,67 @@ public void createSnapshotRowIdsParquetFile(
     executeSynapseQuery(sqlCreateRowIdTable.render());
   }
 
+  public Map<String, Long> createSnapshotParquetFilesByQuery(
+      AssetSpecification assetSpec,
+      UUID snapshotId,
+      String datasetDataSourceName,
+      String snapshotDataSourceName,
+      String translatedQuery,
+      Boolean isGlobalFieldIds)
+      throws SQLException {
+    Map<String, Long> tableRowCounts = new HashMap<>();
+
+    // First handle root table
+    AssetTable rootTable = assetSpec.getRootTable();
+    String rootTableName = rootTable.getTable().getName();
+
+    ST sqlCreateSnapshotTableTemplate = new ST(createSnapshotTableByQueryTemplate);
+    ST queryTemplate =
+        generateSnapshotParquetCreateQuery(
+            sqlCreateSnapshotTableTemplate,
+            IngestUtils.getSourceDatasetParquetFilePath(rootTable.getTable().getName()),
+            rootTable.getTable().getName(),
+            snapshotId,
+            IngestUtils.getSnapshotSliceParquetFilePath(snapshotId, rootTableName, "root"),
+            datasetDataSourceName,
+            snapshotDataSourceName,
+            rootTable.getSynapseColumns(),
+            isGlobalFieldIds);
+
+    queryTemplate.add("query", translatedQuery);
+    int rows;
+    try {
+      rows = executeSynapseQuery(queryTemplate.render());
+    } catch (DataAccessException ex) {
+      throw new PdaoException(
```

Comment:
This catches and rethrows `DataAccessException` but not `SQLException`. The caller in `CreateSnapshotByQueryParquetFilesAzureStep.createSnapshotPrimaryData()` catches `SQLException` but not `PdaoException`. It looks like if this exception occurs, it will result in an uncaught step exception. Is this correct? I would expect this to catch `SQLException` and rethrow as a `PdaoException`, and have the step catch `PdaoException`.

Reply:
I think we want to handle the DataAccessException separately because that is in place to handle the case that the source for the query is empty and provide a meaningful exception to the user. 
Given your suggestion, I've added PdaoException to the catch/throw clauses of the steps.